rct229/rulesets/ashrae9012019/ruleset_functions/get_energy_required_to_heat_swh_use.py

```diff
@@ -0,0 +1,186 @@
+) -> dict[str, Quantity]:
+    """
+    This function calculates the total energy required to heat the SWH use over the course of a year in btu.  Note - this function does not work for service water heating uses with use_units == "OTHER".  In this case, it will return 0.
+
+    Parameters
+    ----------
+    swh_use: dict, service_water_heating_uses
+    rmd: dict, RMD at RuleSetModelDescription level
+    building_segment: dict, building_segment
+
+    Returns
+    ----------
+    energy_required_by_space: A dict where the keys are space_ids and values are the total energy required to heat the swh_use for that space.  If a swh_use is not assigned to any spaces, the key will be "NO_SPACES_ASSIGNED"
+
+    """
+
+    hourly_schedule_id = swh_use.get("use_multiplier_schedule")
+    hourly_schedule = (
+        find_exactly_one_schedule(rmd, hourly_schedule_id)
+        if hourly_schedule_id is not None
+        else None
+    )
+
+    distribution_system_id = swh_use.get("served_by_distribution_system", "")
+    distribution_system = find_exactly_one_with_field_value(
+        "$.service_water_heating_distribution_systems[*]",
+        "id",
+        distribution_system_id,
+        rmd,
+    )
+    supply_temperature = distribution_system.get("design_supply_temperature")
+    assert_(
+        supply_temperature is not None,
+        "A service water heating supply temperature must be provided",
+    )
+    swh_use_value = swh_use.get("use", 0)
+    inlet_temperature_schedule_id = distribution_system.get(
+        "entering_water_mains_temperature_schedule", ""
+    )
+    inlet_temperature_schedule = find_exactly_one_schedule(
+        rmd, inlet_temperature_schedule_id
+    )
+
+    inlet_temperature_hourly_values = inlet_temperature_schedule.get("hourly_values")
+    assert_(
+        inlet_temperature_hourly_values is not None,
+        "A service water heating inlet hourly temperature schedule must be provided",
+    )
+
+    drain_heat_recovery_efficiency = (
+        distribution_system.get("drain_heat_recovery_efficiency")
+        if (swh_use.get("is_heat_recovered_by_drain"))
+        else 0
+    )
+    use_units = swh_use.get("use_units")
+    assert_(use_units is not None, "A use unit must be provided")
+
+    space_ids = get_spaces_served_by_swh_use(rmd, swh_use["id"])
+    space_within_building_segment_ids = [
+        space["id"] for space in find_all("$.zones[*].spaces[*]", building_segment)
+    ]
+    spaces = [
+        find_exactly_one_with_field_value(
+            "$.buildings[*].building_segments[*].zones[*].spaces[*]",
+            "id",
+            space_id,
+            rmd,
+        )
+        for space_id in space_ids
+        if space_id in space_within_building_segment_ids
+    ]
+
+    if len(spaces) == 0 and use_units not in REQUIRED_USE_UNIT:
+        spaces = find_all("$.zones[*].spaces[*]", building_segment)
+
+    energy_required_by_space = {}
+    volume_flow_rate = 0
```

Comment:
This should be ```ZERO.FLOW```.

Reply:
And this would lead to a test failure. The unit of ZERO.FLOW is ft3/min while here it is gallon/h.

Reply:
In that case, I think that you will have to get the swh_use_value multiply by "gallon/hr" to make pint work.
The issue is possibly because you are adding a pint quantity object with a float?